### PR TITLE
Docs 4864/asm user tracking updates

### PR DESCRIPTION
**Pull request by maxhedouin**
### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by hestonhoffman** [comment]
Thanks for the update @maxhedouin, I've created an [Editorial review card](https://datadoghq.atlassian.net/browse/DOCS-4871) and someone on the docs team will take a look soon

**Review by robertpi — APPROVED** [review]

File: content/en/security/application_security/threats/add-user-info.md [hunk 1]
```diff
@@ -15,13 +15,17 @@ further_reading:
 
 ## Overview
 
-Instrument your services with the standardized user tags to track authenticated user activity, whether you're tracking application performance or application security.
+Instrument your services and track user activity to detect and block bad actors.
 
-This way, you can identify bad actors that are generating suspicious security activity, review all their activity around this time frame, and prioritize handling the most advanced attacks and signals targeting your authenticated attack surface.
+Add authenticated user information on traces to identify and block bad actors targeting your authenticated attack surface. The authenticated user monitoring and blocking is done by setting the user id tag on the running Application Performance Monitoring (APM) trace. It also provides the necessary instrumentation for ASM to block authenticated attackers. This allows ASM to associate attacks and business logic events to users.
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:20]:
```suggestion
Add authenticated user information on traces to identify and block bad actors targeting your authenticated attack surface. To do this, set the user ID tag on the running APM trace, providing the necessary instrumentation for ASM to block authenticated attackers. This allows ASM to associate attacks and business logic events to users.
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 2]
```diff
@@ -15,13 +15,17 @@ further_reading:
...
 
-You can [add custom tags to your root span][1], or use the instrumentation functions described below.
+Track user logins to detect account takeovers with out of the box detection rules, and ultimately block attackers.
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:22]:
```suggestion
Track user logins to detect account takeovers with out-of-the-box detection rules, and to ultimately block attackers.
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 3]
```diff
@@ -267,6 +273,384 @@ def view():
 
 {{< /programming-lang-wrapper >}}
 
+## Adding user events (login success, login failure, any business logic) to traces
+
+{{< programming-lang-wrapper langs="java,dotnet,go,ruby,php,nodejs,python" >}}
+{{< programming-lang lang="java" >}}
+
+Use the the Java tracer's API to track user events, released in dd-trace-java v1.8.0.
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:281]:
```suggestion
Starting in dd-trace-java v 1.8.0, you can use the the Java tracer's API to track user events. 
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 4]
```diff
@@ -267,6 +273,384 @@ def view():
...
 {{< /programming-lang-wrapper >}}
 
+## Adding user events (login success, login failure, any business logic) to traces
+
+{{< programming-lang-wrapper langs="java,dotnet,go,ruby,php,nodejs,python" >}}
+{{< programming-lang lang="java" >}}
+
+Use the the Java tracer's API to track user events, released in dd-trace-java v1.8.0. 
+
+The example below shows how to track login events, as well as custom events (using signup as an example)
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:283]:
```suggestion
The following examples show how to track login events or custom events (using signup as an example).
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 5]
```diff
@@ -267,6 +273,384 @@ def view():
...
+
+{{< tabs >}}
+{{% tab "login success" %}}
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:286]:
```suggestion
{{% tab "Login success" %}}
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 6]
```diff
@@ -267,6 +273,384 @@ def view():
...
+            .getEventTracker()
+            .trackLoginSuccessEvent(user.getId(), metadata);
+
+    }
+}
+
+```
+{{% /tab %}}
+
+{{% tab "login failure" %}}
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:311]:
```suggestion
{{% tab "Login failure" %}}
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 7]
```diff
@@ -267,6 +273,384 @@ def view():
...
+        // track user authentication error events
+        GlobalTracer
+            .getEventTracker()
+            .trackLoginFailureEvent(userId, userExists, metadata);
+    }
+}
+```
+{{% /tab %}}
+
+{{% tab "custom business logic" %}}
```
Comment (kayayarai) [inline content/en/security/application_security/threats/add-user-info.md:338]:
```suggestion
{{% tab "Custom business logic" %}}
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
Some minor edits. I'll commit these, and repeat the same edits on all the tabs.

**Review by kayayarai — APPROVED** [review]

File: content/en/security/application_security/threats/add-user-info.md [hunk 8]
```diff
@@ -267,6 +273,390 @@ def view():
...
+{{% /tab %}}
+
+{{< /tabs >}}
+
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="python" >}}
+
+Starting in dd-trace-py v1.9.0, you can use the the Python tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:615]:
```suggestion
Starting in dd-trace-py v1.9.0, you can use the Python tracer's API to track user events.
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 9]
```diff
@@ -267,6 +273,390 @@ def view():
 
 {{< /programming-lang-wrapper >}}
 
+## Adding user events (login success, login failure, any business logic) to traces
+
+{{< programming-lang-wrapper langs="java,dotnet,go,ruby,php,nodejs,python" >}}
+{{< programming-lang lang="java" >}}
+
+Starting in dd-trace-java v1.8.0, you can use the the Java tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:281]:
```suggestion
Starting in dd-trace-java v1.8.0, you can use the Java tracer's API to track user events. 
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 10]
```diff
@@ -267,6 +273,390 @@ def view():
...
+```
+{{% /tab %}}
+
+{{< /tabs >}}
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="dotnet" >}}
+
+Starting in dd-trace-dotnet v2.23.0, you can use the the .NET tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:369]:
```suggestion
Starting in dd-trace-dotnet v2.23.0, you can use the .NET tracer's API to track user events. 
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 11]
```diff
@@ -267,6 +273,390 @@ def view():
...
+```
+{{% /tab %}}
+
+{{< /tabs >}}
+
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
+Starting in dd-trace-go v1.47.0, you can use the the Go tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:434]:
```suggestion
Starting in dd-trace-go v1.47.0, you can use the Go tracer's API to track user events. 
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 12]
```diff
@@ -267,6 +273,390 @@ def view():
...
+}
+```
+{{% /tab %}}
+
+{{< /tabs >}}
+
+{{< /programming-lang >}}
+{{< programming-lang lang="ruby" >}}
+
+Starting in dd-trace-rb v1.9.0, you can use the the Ruby tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:486]:
```suggestion
Starting in dd-trace-rb v1.9.0, you can use the Ruby tracer's API to track user events.
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 13]
```diff
@@ -267,6 +273,390 @@ def view():
...
+# Leveraging custom business logic tracking to track user signups
+Datadog::Kit::AppSec::Events.track('users.signup', trace)
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="php" >}}
+Starting in dd-trace-php v0.84.0, you can use the the PHP tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:527]:
```suggestion
Starting in dd-trace-php v0.84.0, you can use the PHP tracer's API to track user events.
```

File: content/en/security/application_security/threats/add-user-info.md [hunk 14]
```diff
@@ -267,6 +273,390 @@ def view():
...
+?>
+```
+{{% /tab %}}
+
+{{< /tabs >}}
+
+
+{{< /programming-lang >}}
+{{< programming-lang lang="nodejs" >}}
+Starting in dd-trace-js v3.13.1, you can use the the NodeJS tracer's API to track user events.
```
Comment (Julio-Guerra) [inline content/en/security/application_security/threats/add-user-info.md:561]:
```suggestion
Starting in dd-trace-js v3.13.1, you can use the NodeJS tracer's API to track user events.
```

**Pull request merged**